v2/tools/asoctl/internal/importing/importable_arm_resource.go

```diff
@@ -75,24 +90,21 @@ func (i *importableARMResource) Resource() genruntime.MetaObject {
 // Returns a slice of child resources needing to be imported (if any), and/or an error.
 // Both are returned to allow returning partial results in the case of a partial failure.
 func (i *importableARMResource) Import(ctx context.Context) ([]ImportableResource, error) {
-	// Parse ID into a more useful form
-	id, err := arm.ParseResourceID(i.armID)
-	if err != nil {
-		return nil, err // arm.ParseResourceID already returns a good error, no need to wrap
-	}
-
 	var ref genruntime.ResourceReference
-	ref, err = i.importResource(ctx, id)
+	ref, err := i.importResource(ctx, i.armID)
 	if err != nil {
 		return nil, err
 	}
 
 	var result []ImportableResource
-	subTypes := FindSubTypesForType(id.ResourceType.String())
-	for _, subType := range subTypes {
+
+	childTypes := FindChildResourcesForResourceType(i.armID.ResourceType.String())
+	childTypes = append(childTypes, FindExtensionTypes()...)
```

Comment:
minor: Might be worth leaving a comment here saying something like: "Extension types are children of every other type" or something like that?